phoenix-core/src/main/java/org/apache/phoenix/util/ScanUtil.java

```diff
@@ -657,6 +660,22 @@ public static PName padTenantIdIfNecessary(RowKeySchema schema, boolean isSalted
         return tenantId;
     }
 
+    public static byte[] getTenantIdBytes(RowKeySchema schema, boolean isSalted, PName tenantId)
+            throws SQLException {
+        int pkPos = isSalted ? 1 : 0;
+        Field field = schema.getField(pkPos);
+        PDataType dataType = field.getDataType();
+        byte[] convertedValue;
+        try {
+            Object value = dataType.toObject(tenantId.getString());
+            convertedValue = dataType.toBytes(value);
```

Comment:
You can use the following code to pad the convertedValue (this will be a noop for types that aren't padded):

    ImmutableBytesWritable ptr = new ImmutableBytesWritable(convertedValue);
    dataType.pad(ptr, field.getMaxLength(), field.getSortOrder());
    convertedValue = ByteUtil.copyKeyBytesIfNecessary(ptr);